libcudacxx/test/libcudacxx/std/numerics/bit/bit.cast/bit_cast.pass.cpp

```diff
@@ -0,0 +1,348 @@
+    test_roundtrip_through_buffer(i);
+  }
+
+  for (long long i : generate_signed_integral_values<long long>())
+  {
+    test_roundtrip_through_nested_T(i);
+    test_roundtrip_through_buffer(i);
+    test_roundtrip_through<double>(i);
+  }
+
+  // Fundamental unsigned integer types
+  for (unsigned char i : generate_unsigned_integral_values<unsigned char>())
+  {
+    test_roundtrip_through_nested_T(i);
+    test_roundtrip_through_buffer(i);
+  }
+
+  for (unsigned short i : generate_unsigned_integral_values<unsigned short>())
+  {
+    test_roundtrip_through_nested_T(i);
+    test_roundtrip_through_buffer(i);
+  }
+
+  for (unsigned int i : generate_unsigned_integral_values<unsigned int>())
+  {
+    test_roundtrip_through_nested_T(i);
+    test_roundtrip_through_buffer(i);
+    test_roundtrip_through<float>(i);
+  }
+
+  for (unsigned long i : generate_unsigned_integral_values<unsigned long>())
+  {
+    test_roundtrip_through_nested_T(i);
+    test_roundtrip_through_buffer(i);
+  }
+
+  for (unsigned long long i : generate_unsigned_integral_values<unsigned long long>())
+  {
+    test_roundtrip_through_nested_T(i);
+    test_roundtrip_through_buffer(i);
+    test_roundtrip_through<double>(i);
+  }
+
+  // Fixed width signed integer types
+  for (cuda::std::int32_t i : generate_signed_integral_values<cuda::std::int32_t>())
+  {
+    test_roundtrip_through_nested_T(i);
+    test_roundtrip_through_buffer(i);
+    test_roundtrip_through<int>(i);
+    test_roundtrip_through<cuda::std::uint32_t>(i);
+    test_roundtrip_through<float>(i);
+  }
+
+  for (cuda::std::int64_t i : generate_signed_integral_values<cuda::std::int64_t>())
+  {
+    test_roundtrip_through_nested_T(i);
+    test_roundtrip_through_buffer(i);
+    test_roundtrip_through<long long>(i);
+    test_roundtrip_through<cuda::std::uint64_t>(i);
+    test_roundtrip_through<double>(i);
+  }
+
+  // Fixed width unsigned integer types
+  for (cuda::std::uint32_t i : generate_unsigned_integral_values<cuda::std::uint32_t>())
+  {
+    test_roundtrip_through_nested_T(i);
+    test_roundtrip_through_buffer(i);
+    test_roundtrip_through<int>(i);
+    test_roundtrip_through<cuda::std::int32_t>(i);
+    test_roundtrip_through<float>(i);
+  }
+
+  for (cuda::std::uint64_t i : generate_unsigned_integral_values<cuda::std::uint64_t>())
+  {
+    test_roundtrip_through_nested_T(i);
+    test_roundtrip_through_buffer(i);
+    test_roundtrip_through<long long>(i);
+    test_roundtrip_through<cuda::std::int64_t>(i);
+    test_roundtrip_through<double>(i);
+  }
+
+  // Floating point types
+  for (float i :
+       {0.0f,
+        1.0f,
+        -1.0f,
+        10.0f,
+        -10.0f,
+        1e10f,
+        1e-10f,
+        1e20f,
+        1e-20f,
+        2.71828f,
+        3.14159f,
+        cuda::std::nanf(""),
+        __builtin_nanf("0x55550001"), // NaN with a payload
+        cuda::std::numeric_limits<float>::signaling_NaN(),
+        cuda::std::numeric_limits<float>::quiet_NaN()})
```

Comment:
Suggestion: since you are so rigorously testing various floats, let's throw in infinity as well.